Makefile

```diff
@@ -167,3 +169,84 @@ endif
 # Help documentatin à la https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
 help:
 	@cat Makefile | grep -v '\.PHONY' |  grep -v '\help:' | grep -B1 -E '^[a-zA-Z_.-]+:.*' | sed -e "s/:.*//" | sed -e "s/^## //" |  grep -v '\-\-' | sed '1!G;h;$$!d' | awk 'NR%2{printf "\033[36m%-30s\033[0m",$$0;next;}1' | sort
+
+# server-debug builds and deploys a debug version of the plugin for your architecture.
+# Then resets the plugin to pick up the changes.
+.PHONY: debug
+debug: server-debug reset
+
+.PHONY: server-debug
+server-debug:
+
+	./build/bin/manifest apply
+	mkdir -p server/dist
+
+ifeq ($(OS),Darwin)
+	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-darwin-amd64;
+else ifeq ($(OS),Linux)
+	cd server && env GOOS=linux GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-linux-amd64;
+else ifeq ($(OS),Windows_NT)
+	cd server && env GOOS=windows GOARCH=amd64 $(GO) build -gcflags "all=-N -l" -o dist/plugin-windows-amd64.exe;
+else
+	$(error make debug depends on uname to return your OS. If it does not return 'Darwin' (meaning OSX), 'Linux', or 'Windows_NT' (all recent versions of Windows), you will need to edit the Makefile for your own OS.)
+endif
+
+	rm -rf dist/
+	mkdir -p dist/$(PLUGIN_ID)/server/dist
+	cp $(MANIFEST_FILE) dist/$(PLUGIN_ID)/
+	cp -r server/dist/* dist/$(PLUGIN_ID)/server/dist/
+	mkdir -p ../mattermost-server/plugins
```

Comment:
The problem with using deploy semantics is the symlink to the dist folder... As far as I can tell, when a plugin is installed via the API, it calls os.RemoveAll(pluginPath).... Which would be fine, if we just went in and deleted the webapp/dist folder and relinked it. But then we'd have to rely on paths anyway.

If we want the webapp watcher benefits and the speed of just calling `make reset` for a webapp change, then I think we have to link the directories. So I'm okay with making it a requirement to develop the plugin in a peer directory. If someone doesn't want to, then they're free to put it wherever they want (they just can't use these make commands). 

I added a check and it will error out when its not a peer to the server. That should help new devs figure out what's wrong, and experienced devs who know what they're doing will decide if the tradeoff is worth it for them.

Reply:
Maybe we can offer a environment variable that devs have to set similar to what we do with `MM_SERVICESETTINGS_SITEURL`? The default would be look for the server at ` ../mattermost-server/`, but if the the environment variable is set, we use this instead.